packages/editor/src/components/dialogs/PublishModal.tsx

```diff
@@ -0,0 +1,271 @@
+/*
+CPAL-1.0 License
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/EtherealEngine/etherealengine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+The Original Code is Ethereal Engine.
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { NotificationService } from '@etherealengine/client-core/src/common/services/NotificationService'
+import { PopoverState } from '@etherealengine/client-core/src/common/services/PopoverState'
+import {
+  LocationData,
+  LocationID,
+  locationPath,
+  LocationType,
+  staticResourcePath
+} from '@etherealengine/common/src/schema.type.module'
+import { useHookstate } from '@etherealengine/hyperflux'
+import { useFind, useMutation } from '@etherealengine/spatial/src/common/functions/FeathersHooks'
+import Button from '@etherealengine/ui/src/primitives/tailwind/Button'
+import Input from '@etherealengine/ui/src/primitives/tailwind/Input'
+import LoadingView from '@etherealengine/ui/src/primitives/tailwind/LoadingView'
+import { ModalHeader } from '@etherealengine/ui/src/primitives/tailwind/Modal'
+import Select from '@etherealengine/ui/src/primitives/tailwind/Select'
+import Toggle from '@etherealengine/ui/src/primitives/tailwind/Toggle'
+import { HiLink } from 'react-icons/hi2'
+
+const getDefaultErrors = () => ({
+  name: '',
+  maxUsers: '',
+  scene: '',
+  serverError: ''
+})
+
+const locationTypeOptions = [
+  { label: 'Private', value: 'private' },
+  { label: 'Public', value: 'public' },
+  { label: 'Showroom', value: 'showroom' }
+]
+
+export default function PublishSceneModal({ location, sceneID }: { location?: LocationType; sceneID: string | null }) {
```

Comment:
@HexaField do you think we need a separate control for this or `AddEditLocationModal` should be used?

Reply:
If it's similar enough we should reuse AddEditLocationModal

Reply:
done